core/src/miner/mem_pool.rs

```diff
@@ -611,45 +525,139 @@ impl MemPool {
+            if let Some(old_order_with_tag) = self.by_signer_public.insert(signer_public, seq, order_with_tag) {
+                let old_order = old_order_with_tag.order;
+                let tag = old_order_with_tag.tag;
+
+                self.by_hash.remove(&old_order.hash);
+                match tag {
+                    QueueTag::Current => {
+                        self.current.remove(&old_order);
+                    }
+                    QueueTag::Future => {
+                        self.future.remove(&old_order);
+                    }
+                    QueueTag::New => unreachable!(),
+                }
+            }
+
+            to_insert.entry(signer_public).or_default().push(seq);
+            insert_results.push(Ok((signer_public, seq)));
+        }
+
+        let keys = self.by_signer_public.keys().map(Clone::clone).collect::<Vec<_>>();
+
+        for public in keys {
+            let current_seq = fetch_account(&public).seq;
+            let mut first_seq = *self.first_seqs.get(&public).unwrap_or(&0);
+            let next_seq = self.next_seqs.get(&public).cloned().unwrap_or(current_seq);
+
+            let new_next_seq = if current_seq < first_seq
+                || current_time < self.last_time
+                || current_timestamp < self.last_timestamp
+                || next_seq < current_seq
+            {
+                self.check_transactions(public, current_seq, current_time, current_timestamp)
+            } else {
+                to_insert
+                    .get(&public)
+                    .and_then(|v| self.check_new_transactions(public, v, next_seq, current_time, current_timestamp))
+                    .unwrap_or_else(|| self.check_transactions(public, next_seq, current_time, current_timestamp))
+            };
+
+            // Need to update the height
+            if current_seq != first_seq {
+                self.update_seq(public, current_seq, new_next_seq);
+                self.first_seqs.insert(public, current_seq);
+                first_seq = current_seq;
+            }
+            // We don't need to update the height, just move transacitons
+            else if new_next_seq < next_seq {
+                self.move_queue(public, new_next_seq, next_seq, QueueTag::Future);
+            } else if new_next_seq > next_seq {
+                self.move_queue(public, next_seq, new_next_seq, QueueTag::Current);
+            }
+
+
+            if new_next_seq <= first_seq {
+                self.next_seqs.remove(&public);
+            } else {
+                self.next_seqs.insert(public, new_next_seq);
+            }
+
+            if let Some(seq_list) = to_insert.get(&public) {
+                self.add_new_orders_to_queue(public, seq_list, new_next_seq);
+            }
+
+            self.by_signer_public.clear_if_empty(&public);
+        }
+
+        // FIXME: Enforcing limit too late?
+        // The logic above will allow memory usage more than the limit for a moment.
```

Comment:
I think it's okay.
Actually, we don't enforce that the mempool uses less than the limitation because of the local transactions.

Reply:
For the same reason, I think it's okay not to call `enforce_limit` in `set_limit`.

Reply:
But the original behavior is to call `enforce_limit`. I'm not sure...
+ I forgot to not to include local/retracted transactions to the memory limit. I'll fix this.